src/Compilers/CSharp/Portable/Syntax/SyntaxNodeExtensions.cs

```diff
@@ -293,5 +293,15 @@ private static bool IsDeconstructionCompatibleArgument(ExpressionSyntax expressi
 
             return false;
         }
+
+        internal static Location GetLocationForDiagnostics(this SyntaxNode node)
+        {
+            if (node is LambdaExpressionSyntax lambda)
+            {
+                return Location.Create(lambda.SyntaxTree, Text.TextSpan.FromBounds(lambda.SpanStart, lambda.ArrowToken.Span.End));
```

Comment:
>return Location.Create(lambda.SyntaxTree, Text.TextSpan.FromBounds(lambda.SpanStart, lambda.ArrowToken.Span.End)); [](start = 16, length = 114)

We already have LambdaSymbol.DiagnosticLocation property which has different implementation and handles ```delegate ...``` syntax. Perhaps we should unify/share/reuse the implementation.